pkg/extensions/search/resolver.go

```diff
@@ -279,6 +279,260 @@ func getCVEListForImage(
+						Repo:   baseImage.Repo,
+						Tag:    baseImage.Tag,
+						Digest: ref(baseImage.Manifests[0].Digest.String()),
+						Platform: &gql_generated.Platform{
+							Os:   ref(baseImage.Manifests[0].Config.OS),
+							Arch: ref(getArch(baseImage.Manifests[0].Config.Platform)),
+						},
+					}
+					subtrahend.Repo = baseImage.Repo
+					subtrahend.Tag = baseImage.Tag
+					subtrahend.Digest = ref(baseImage.Manifests[0].Digest.String())
+				}
+			}
+		}
+	}
+
+	minuendRepoRef := minuend.Repo + "@" + deref(minuend.Digest, "")
+	subtrahendRepoRef := subtrahend.Repo + "@" + deref(subtrahend.Digest, "")
+	page := dderef(requestedPage)
+
+	diffCVEs, diffSummary, _, err := cveInfo.GetCVEDiffListForImages(ctx, minuendRepoRef, subtrahendRepoRef, searchedCVE,
+		excludedCVE, cvemodel.PageInput{
+			Limit:  deref(page.Limit, 0),
+			Offset: deref(page.Offset, 0),
+			SortBy: cvemodel.SortCriteria(deref(page.SortBy, gql_generated.SortCriteriaSeverity)),
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	cveids := []*gql_generated.Cve{}
+
+	for _, cveDetail := range diffCVEs {
+		vulID := cveDetail.ID
+		desc := cveDetail.Description
+		title := cveDetail.Title
+		severity := cveDetail.Severity
+		referenceURL := cveDetail.Reference
+
+		pkgList := make([]*gql_generated.PackageInfo, 0)
+
+		for _, pkg := range cveDetail.PackageList {
+			pkg := pkg
+
+			pkgList = append(pkgList,
+				&gql_generated.PackageInfo{
+					Name:             &pkg.Name,
+					InstalledVersion: &pkg.InstalledVersion,
+					FixedVersion:     &pkg.FixedVersion,
+				},
+			)
+		}
+
+		cveids = append(cveids,
+			&gql_generated.Cve{
+				ID:          &vulID,
+				Title:       &title,
+				Description: &desc,
+				Severity:    &severity,
+				Reference:   &referenceURL,
+				PackageList: pkgList,
+			},
+		)
+	}
+
+	return &gql_generated.CVEDiffResult{
+		Minuend:    &resultMinuend,
+		Subtrahend: &resultSubtrahend,
+		Summary: &gql_generated.ImageVulnerabilitySummary{
+			Count:         &diffSummary.Count,
+			UnknownCount:  &diffSummary.UnknownCount,
+			LowCount:      &diffSummary.LowCount,
+			MediumCount:   &diffSummary.MediumCount,
+			HighCount:     &diffSummary.HighCount,
+			CriticalCount: &diffSummary.CriticalCount,
+			MaxSeverity:   &diffSummary.MaxSeverity,
+		},
+		CVEList: cveids,
+		Page:    &gql_generated.PageInfo{},
+	}, nil
+}
+
+func getImageIdentifier(img gql_generated.ImageInput) gql_generated.ImageIdentifier {
+	return gql_generated.ImageIdentifier{
+		Repo:     img.Repo,
+		Tag:      img.Tag,
+		Digest:   img.Digest,
+		Platform: getIdentifierPlatform(img.Platform),
+	}
+}
+
+func getIdentifierPlatform(platform *gql_generated.PlatformInput) *gql_generated.Platform {
+	if platform == nil {
+		return nil
+	}
+
+	return &gql_generated.Platform{
+		Os:   platform.Os,
+		Arch: platform.Arch,
+	}
+}
+
+// rename idea: identify image from input.
+func checkImageInput(ctx context.Context, imageInput gql_generated.ImageInput, metaDB mTypes.MetaDB,
```

Comment:
Definitely needs a rename.
 
```suggestion
func resolveImageData(ctx context.Context, imageInput gql_generated.ImageInput, metaDB mTypes.MetaDB,
```

Or

```suggestion
func resolveImageDigest(ctx context.Context, imageInput gql_generated.ImageInput, metaDB mTypes.MetaDB,
```